test/e2e/clustertask/start_test.go

```diff
@@ -47,47 +51,55 @@ func TestClusterTaskInteractiveStartE2E(t *testing.T) {
 	kubectl := cli.NewKubectl(namespace)
 	tkn, err := cli.NewTknRunner(namespace)
 	assert.NilError(t, err)
+	// Set environment variable TEST_CLUSTERTASK_LIST_EMPTY to any value to skip "No ClusterTasks found" test
+	if os.Getenv("TEST_CLUSTERTASK_LIST_EMPTY") == "" {
+		t.Run("Get list of ClusterTasks when none present", func(t *testing.T) {
+			res := tkn.Run("clustertask", "list")
+			expected := "No ClusterTasks found\n"
+			res.Assert(t, icmd.Expected{
+				ExitCode: 0,
+				Err:      icmd.None,
+				Out:      expected,
+			})
 
-	t.Run("Get list of ClusterTasks when none present", func(t *testing.T) {
-		res := tkn.Run("clustertask", "list")
-		expected := "No ClusterTasks found\n"
-		res.Assert(t, icmd.Expected{
-			ExitCode: 0,
-			Err:      icmd.None,
-			Out:      expected,
 		})
-	})
-
+	}
 	t.Logf("Creating clustertask read-clustertask")
-	kubectl.MustSucceed(t, "create", "-f", helper.GetResourcePath("read-file-clustertask.yaml"))
+	res := kubectl.MustSucceed(t, "create", "-f", helper.GetResourcePath("read-file-clustertask.yaml"))
+	regex := regexp.MustCompile(`read-clustertask-[a-z0-9]+`)
+	clusterTaskName := regex.FindString(res.Stdout())
 
 	t.Logf("Creating git pipeline resource in namespace: %s", namespace)
 	kubectl.MustSucceed(t, "create", "-f", helper.GetResourcePath("git-resource.yaml"))
 
 	t.Run("Get list of ClusterTasks", func(t *testing.T) {
 		res := tkn.Run("clustertask", "list")
-		expected := builder.ListAllClusterTasksOutput(t, c, map[int]interface{}{
-			0: &builder.TaskData{
-				Name: "read-clustertask",
-			},
-		})
-		res.Assert(t, icmd.Expected{
-			ExitCode: 0,
-			Err:      icmd.None,
-			Out:      expected,
-		})
+		if os.Getenv("TEST_CLUSTERTASK_LIST_EMPTY") == "" {
```

Comment:
Why is this check needed?

Reply:
If the cluster contains other clustertasks, this test fails because we are checking for only one clustertask (`read-clustertask`) in the output of ``tkn ct ls`. So adding a check if